CSharp/Library/Microsoft.Bot.Connector.Shared/ServiceProvider.cs

```diff
@@ -99,5 +107,49 @@ private static void ThrowOnNullInstance()
 
             return service;
         }
+
+        /// <summary>
+        /// Tries to auto register the ASP.NET implementation of Connector for backward compatibility.
+        /// </summary>
+        private static void TryAutoRegisterForBackwardCompatibility()
+        {
+            const string AspNetBotConnectorAssemblyName = "Microsoft.Bot.Connector";
+            const string AspNetBotConnectorServiceProviderFullQualifiedName = "Microsoft.Bot.Connector.BotServiceProvider";
+
+            Assembly connectorAssembly = null;
+
+            try
+            {
+                var connectorAssemblyName = new AssemblyName(AspNetBotConnectorAssemblyName);
```

Comment:
Yes. I ended up testing it to make sure. Assembly.Load doesn't require the token key when loading signed binaries. We could add it, though, to prevent loading a unsigned version of the assembly, but I'm afraid it would prevent any one that doesn't have the key to work unsigned builds.